Compatibility-2.0.md

```diff
@@ -26,3 +26,304 @@ Embedded Mode also might use a decorated Cache using a Transcoder. For example t
 We must ensure that Consistent-Hash-aware clients knowledge of the hashing of keys is not affected by transcoding.
 
 Transcoders should be deployable
+
+
+# Encoders, Transcoders and indexable Json 
+
+[ISPN-7896](https://issues.jboss.org/browse/ISPN-7896) introduced the Encoders, adding capability to the caches for on-demand data conversion instead of relying on implicit conversions based on a combination of flags and interceptors. 
+
+This section will expand on the Encoders usage and will introduce the complementary Transcoders, used to convert between different encodings.
+
+## About Encoders
+
+An Encoder is essentially a two-way conversion between the storage format in a cache and the read/write format. The ```AdvancedCache```  interface has a method to specify an encoder to be used during all operations involving that cache:
+
+```cache.getAdvancedCache().withEncoding(MyEncoder.class)```
+
+which returns a decorated cache instance where all operations (read/write/stream/add listener, etc) flow through the supplied encoder.
+
+At the moment, there is no cache configuration to specify an Encoder; Infinispan tries to pick the best decorated cache depending on the
+situation where this cache is used. For example:
+
+* Embedded caches with ```Storage.OBJECT``` are associated with the ```IdentityEncoder```, which stores objects "as is";
+* Embedded caches with ```Storage.OFF_HEAP``` are associated with a ```MarshallerEncoder```, which marshalls data before storing in order to produce a ```byte[]``` (hard requirement);
+* Embedded caches with ```Storage.BINARY``` are associated with the BinaryEncoder, which produces a mix of marshalled and unmarshalled objects.
```

Comment:
This is describing how ```storeAsBinary``` works: it marshalls all pojos but store primitives and Strings as is